components/brave_shields/core/browser/ad_block_component_service_manager.cc

```diff
@@ -159,12 +162,15 @@ void AdBlockComponentServiceManager::UpdateFilterListPrefs(
   if (!local_state_) {
     return;
   }
-  ScopedDictPrefUpdate update(local_state_, prefs::kAdBlockRegionalFilters);
-  base::Value::Dict regional_filter_dict;
-  regional_filter_dict.Set("enabled", enabled);
-  update->Set(uuid, std::move(regional_filter_dict));
+  {
+    ScopedDictPrefUpdate update(local_state_, prefs::kAdBlockRegionalFilters);
+    base::Value::Dict regional_filter_dict;
+    regional_filter_dict.Set("enabled", enabled);
+    update->Set(uuid, std::move(regional_filter_dict));
+  }
 
   RecordP3ACookieListEnabled();
+  list_p3a_->ReportFilterListUsage();
```

Comment:
why do we have two separate p3a calls here done in two completely different ways? Can't we consolidate these into  an observer method and not include the p3a code in here at all? See my other comment about injecting P3A into everything which doesn't seem great to me

Reply:
the `UMA` stuff is inside the code everywhere, but at least it's always called in a consistent way